Scripts/Extensions/TransformExtensions.cs

```diff
@@ -26,20 +26,22 @@ public static Bounds TransformBounds(this Transform transform, Bounds localBound
         public static Vector3 XZForward(this Transform target)
         {
             var forward = target.forward;
-            if ((forward.y * forward.y) >= 0.5f)
+            var up = forward.y;
+            const float mid = 0.5f;
+            const float min = 0.0f;
+            if ((up * up) >= mid)
             {
-                forward = -target.up * Mathf.Sign(forward.y);
+                forward = -target.up * Mathf.Sign(up);
             }
-            else
+            else if (target.up.y < min)
             {
-                if (target.up.y < 0.0f)
-                {
-                    forward = -forward;
-                }
+                forward = -forward;
             }
-            forward.y = 0.0f;
+
+            up = 0.0f;
```

Comment:
Not sure if this should have changed from forward.y.  Setting up to any value here does nothing.